src/systems/filecoin_markets/deal/deal.id

```diff
@@ -8,17 +8,25 @@ type DealCID ipld.CID
 type Signature struct {}  // TODO
 
 type StorageDealProposal struct {
-    PieceRef            DealCID
-    PieceSize           UInt
-    Client              addr.Address
-    Provider            addr.Address
-    ProposalExpiration  block.ChainEpoch
-    DealExpiration      block.ChainEpoch
-    StoragePrice        actor.TokenAmount
-    StorageCollateral   actor.TokenAmount
-    ProposerSignature   Signature
+    // TODO: Decide if PieceRef needs to go on chain
+    // PieceRef is the hash of the data in native structure. This will be used for
+    // certifying the data transfer. Reference for transit.
+    PieceRef                DealCID
```

Comment:
I *think* the outcome of https://gist.github.com/jbenet/8af91b75091f05a1c92a774f9d9a56c8 is that only CommP needs to go on-chain. Words are tricky here (I don't know what "native" means), but I don't think the CID of the IPLD graph closest to the raw data the client wants to store needs to go on chain. CommP is implicitly coupled to it anyway, since the CommP is calculated by serializing the graph (into a CAR file) then creating the merkle tree from those byptes.

Reply:
A root CID of the not-a-merkle-tree IPLD graph may be transmitted to the miner as part of deal negotiation, but not inside this structure.

Reply:
cc @hannahhoward @jbenet

Reply:
Agreed — those decisions are more recent and supplant what appears here. @jbenet will be cleaning these data structures up soon.

Reply:
This needs to be:
```go
    PieceCID piece.PieceCID // CommP`
```

Reply:
This one needs to be removed -- not needed. (we cannot fit it on chain)